url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/representation/BitboardPosition.java

```diff
@@ -0,0 +1,2907 @@
+      }
+      //Se comprueba si es una captura de una torre en casilla
+      //origigen quitar permiso enroque.
+      if(dp != null && dp.genericPiece == GenericPiece.ROOK)
+      {
+        if(ds == Square.a1 && this.whiteCastlingLong)
+        {
+           this.setCastlingLong(Colour.WHITE, false);
+        }
+        else if(ds == Square.a8 && this.blackCastlingLong)
+        {
+           this.setCastlingLong(Colour.BLACK, false);
+        }
+        else if(ds == Square.h1 && this.whiteCastlingShort)
+        {
+           this.setCastlingShort(Colour.WHITE, false);
+        }
+        else if(ds == Square.h8 && this.blackCastlingShort)
+        {
+           this.setCastlingShort(Colour.BLACK, false);
+        }
+      }
+      
+      
+      //Todo ver si el movimiento hace que un peón se quede en disposición
+      //de ser capturado al paso
+      this.setEnPassantSquare(null);
+      if(sp.genericPiece == GenericPiece.PAWN)
+      {
+         if(sp == Piece.WHITE_PAWN)
+         {
+            if(ss.getRank() == Rank._2 && ds.getRank() == Rank._4)
+            {
+               this.setEnPassantSquare(enPassantSquareDestination[ds.index]);
+            }
+         }
+         else
+         {
+            if(ss.getRank() == Rank._7 && ds.getRank() == Rank._5)
+            {
+               this.setEnPassantSquare(enPassantSquareDestination[ds.index]);
+            }
+         }
+      }
+      
+      //Actualizar regla 50 movimientos
+      if(dp != null || sp.genericPiece == GenericPiece.PAWN || mov.isInPassant())
+      {
+         //Si es un movimiento de peón o captura se resetea el contador
+         //de la regla de los 50 movimientos
+         this.fiftyMovesRuleIndex = 0;
+      }
+      else
+      {
+         //Si no se incrementa el contador
+         this.fiftyMovesRuleIndex++;
+      }
+      
+      //TODO Actualizar número de movimiento
+      
+      
+      //Actualizar turno
+      setTurn(turn.opposite());
+      
+      //Se añade la posición resultante en el hash de posiciones
+      Integer keyValue = gamePositionsHash.get(zobristKey.getKey());
+      if(keyValue != null)
+      {
+         gamePositionsHash.put(zobristKey.getKey(), keyValue + 1);
+      }
+      else
+      {
+         gamePositionsHash.put(zobristKey.getKey(), 1);
+      }
+      
+      //Se actualizan los bitboards para la posición resultante
+      updateAttackedFromFull();
+      
+      return true;
+   }
+   
+   /* (non-Javadoc)
+    * @see es.dagaren.gladiator.representation.Position#getMoves(java.util.List)
+    */
+   @Override
+   public void getMoves(List<Movement> movesList)
+   {
+      generarMovements(movesList);
+   }
+   
+   public List<Movement> getMovements()
+   {
+      if(movesGenerated == false)
+      {
+         this.movesList = new ArrayList<Movement>();
+         generarMovements(this.movesList);
+      }
+      
+      return movesList;
+   }
+   
+   public void generarMovements(List<Movement> movesList)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 102. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.